### PR TITLE
Rename pkg/{tagging,lifecycle} to pkg/bucket sub-directory

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Rename pkg/{tagging,lifecycle} to pkg/bucket sub-directory

## Motivation and Context
Rename to allow for more such features to come in a more
proper hierarchical manner.

## How to test this PR?
Nothing to test really, just import paths have changed.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#f45cf94c-b547-4588-b1a0-9e219f6a7afb) |
| mint-fs.sh | :x: [more...](#e0d29869-450f-4ca2-b42c-627906a682ee) |
| mint-gateway-azure.sh | :x: [more...](#12f8c29d-aee1-41b5-842e-c74ae99afbac) |
<a name="f45cf94c-b547-4588-b1a0-9e219f6a7afb"></a>
### 8892-b9e5986/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:30595
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 822251acbc44:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 22 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 6 seconds
(4/13) Running mc tests ... done in 51 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 3 seconds
{
  "args": {},
  "duration": 69589,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="12f8c29d-aee1-41b5-842e-c74ae99afbac"></a>
### 8892-b9e5986/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30254
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 83cfae29cdf9:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 13 seconds
{"name": "awscli", "duration": 11660, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://72.28.97.51:30254/awscli-mint-test-bucket-688""}

Executed 0 out of 13 tests successfully.
```
<a name="e0d29869-450f-4ca2-b42c-627906a682ee"></a>
### 8892-b9e5986/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32463
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d1a36bea566f:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 9 seconds
{"name": "awscli", "duration": 307386, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://72.28.97.55:32463/awscli-mint-test-bucket-15368""}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/api-errors.go [hunk 1]
```diff
@@ -32,11 +32,11 @@ import (
 	"github.com/minio/minio/cmd/crypto"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/pkg/auth"
+	"github.com/minio/minio/pkg/bucket/objectlock"
+	"github.com/minio/minio/pkg/bucket/policy"
+	"github.com/minio/minio/pkg/bucket/tagging"
```
Comment (nitisht) [inline cmd/api-errors.go]: Tagging is technically objectTagging. Can we name it as `pkg/object/tagging`?

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#b70deb9b-bd61-4499-8943-d13970ffc364) |
| mint-large-bucket.sh | :x: [more...](#13e15322-d92f-4865-ba7f-5b9e19956b5a) |
| mint-dist-xl.sh | :x: [more...](#a0fa2302-ec66-4aea-9524-a8ee37263685) |
| mint-gateway-azure.sh | :x: [more...](#c1f2b605-ce36-4de5-9566-024a83ae6d0f) |
<a name="13e15322-d92f-4865-ba7f-5b9e19956b5a"></a>
### 8892-90aa945/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30789
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 27a4ba055d70:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 23 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 6 seconds
(4/13) Running mc tests ... done in 52 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 40 seconds
{
  "args": {},
  "duration": 109785,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="b70deb9b-bd61-4499-8943-d13970ffc364"></a>
### 8892-90aa945/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30187
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0efabe291576:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 10 seconds
{"name": "awscli", "duration": 307947, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://72.28.97.55:30187/awscli-mint-test-bucket-17105""}

Executed 0 out of 13 tests successfully.
```
<a name="c1f2b605-ce36-4de5-9566-024a83ae6d0f"></a>
### 8892-90aa945/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:32150
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1e206978e517:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 14 seconds
{"name": "awscli", "duration": 12448, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://72.28.97.59:32150/awscli-mint-test-bucket-30689""}

Executed 0 out of 13 tests successfully.
```
<a name="a0fa2302-ec66-4aea-9524-a8ee37263685"></a>
### 8892-90aa945/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:31838
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 46a26eaead30:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 12 seconds
{"name": "awscli", "duration": 309248, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://72.28.97.55:31838/awscli-mint-test-bucket-15273""}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/api-errors.go (hunk 1 above)
Comment (poornas) [reply cmd/api-errors.go]: Objectlock should also be under pkg/object/lock since locking is a object level feature

Comment (harshavardhana) [reply cmd/api-errors.go]: Yes but I prefer to keep bucket as a primary top level directory. It is not a representation of API but more about how these features are used for objects or bucket on a bucket.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#28c2895c-2366-4396-8e2d-b128f2db5619) |
| mint-dist-xl.sh | :x: [more...](#40335cc1-e2ae-473c-8980-aec1a9e7368a) |
| mint-gateway-azure.sh | :x: [more...](#2cc78c48-736a-4698-967b-fdce81c5caf3) |
<a name="2cc78c48-736a-4698-967b-fdce81c5caf3"></a>
### 8892-79a6a82/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:30856
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 45f6b4fcf609:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 8 seconds
(2/13) Running aws-sdk-java tests ... done in 2 seconds
(3/13) Running aws-sdk-ruby tests ... done in 16 seconds
(4/13) Running mc tests ... done in 7 minutes and 54 seconds
(5/13) Running minio-go tests ... FAILED in 12 minutes and 17 seconds
{
  "args": {
    "bucketName": "minio-go-test-sn1zctzvuyddnkay",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 233,
  "error": "The specified bucket is not valid.",
  "function": "ListObjects(bucketName, objectPrefix, recursive, doneCh)",
  "message": "PutObject 1 call failed",
  "name": "minio-go: testListObjects",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="28c2895c-2366-4396-8e2d-b128f2db5619"></a>
### 8892-79a6a82/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:30562
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8771223d168a:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 1 minutes and 43 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 16 seconds
(4/13) Running mc tests ... done in 1 minutes and 3 seconds
(5/13) Running minio-go tests ... FAILED in 4 minutes and 22 seconds
{
  "args": {},
  "duration": 110146,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="40335cc1-e2ae-473c-8980-aec1a9e7368a"></a>
### 8892-79a6a82/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31504
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5e6fe223debb:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 12 seconds
{
  "name": "awscli",
  "duration": 7541,
  "function": "aws --endpoint-url http://72.28.97.51:31504 s3api get-object --bucket awscli-mint-test-bucket-16321 --key datafile-1-kB /tmp/datafile-1-kB\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the GetObject operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#80267fae-3324-4dd8-9032-917ca398ad9d) |
| mint-gateway-azure.sh | :x: [more...](#fa72290f-c801-4244-91a8-2ffbf5fabc15) |
<a name="80267fae-3324-4dd8-9032-917ca398ad9d"></a>
### 8892-b0fc0ad/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31545
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 309fa305e654:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 24 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 7 seconds
(4/13) Running mc tests ... done in 1 minutes and 3 seconds
(5/13) Running minio-go tests ... FAILED in 4 minutes and 0 seconds
{
  "args": {},
  "duration": 122736,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="fa72290f-c801-4244-91a8-2ffbf5fabc15"></a>
### 8892-b0fc0ad/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30425
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3fea032170ae:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 23 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 16 seconds
(4/13) Running mc tests ... done in 6 minutes and 50 seconds
(5/13) Running minio-go tests ... FAILED in 10 minutes and 12 seconds
{
  "args": {
    "bucketName": "minio-go-test-serqu0bek51xrdz4",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 234,
  "error": "The specified bucket is not valid.",
  "function": "ListObjects(bucketName, objectPrefix, recursive, doneCh)",
  "message": "PutObject 1 call failed",
  "name": "minio-go: testListObjects",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by nitisht — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Pull request merged**